onnx/defs/math/defs.cc

```diff
@@ -3412,19 +3438,47 @@ bool BuildContextDependentFunctionBodySCE(
       {{"axes"},
        "Constant",
        {},
-       {MakeAttribute("value", ToDimensionOneInt64Tensor(1))}});
+       {MakeAttribute("value", ToDimensionOneInt64Tensor(-1))}});
 
+  // Using stable implementation of LogSoftmax
+  // We need to unsqueeze the inputs on the last axis if they are 2D in order to make the 
+  // further Reshape and transpose valid.
+  body.push_back(
+      {{"X_unsqueezed"},
+        "Unsqueeze",
+       {"scores", "axes"}});
+  body.push_back(
+      {{"3D_shape"},
```

Comment:
Suggest renaming "3D_shape" to a valid C identifier, eg., "ThreeD_shape" or "Shape3D".